libraries/abstractions/common_io/test/test_iot_sdio.c

```diff
@@ -0,0 +1,1042 @@
+    }
+
+    /* Set data bus width */
+    lRetVal = iot_sdio_ioctl(xSdioHandle,
+                             eSDIOSetDataBusWidth,
+                             (void *)&etestIotSdioBusWidth);
+    TEST_ASSERT_EQUAL(IOT_SDIO_SUCCESS, lRetVal);
+
+    /* Verify data bus width */
+    lRetVal = iot_sdio_io_read_direct(xSdioHandle,
+                                eSdioFunction0,
+                                testIotSdio_BUS_CTL_ADDR,
+                                (uint8_t *)&ucReadVal);
+    TEST_ASSERT_EQUAL(IOT_SDIO_SUCCESS, lRetVal);
+    TEST_ASSERT_EQUAL(etestIotSdioBusWidth, testIotSdio_BUS_WIDTH_MASK & ucReadVal);
+
+    /* Select function for suspend/resume */
+    eFunc = eSdioFunction1;
+    lRetVal = iot_sdio_ioctl(xSdioHandle,
+                             eSDIOSelectIo,
+                             (void *)&eFunc);
+    TEST_ASSERT_EQUAL(IOT_SDIO_SUCCESS, lRetVal);
+
+    /* Verify select function */
+    lRetVal = iot_sdio_io_read_direct(xSdioHandle,
+                                eSdioFunction0,
+                                testIotSdio_FUNC_SEL_ADDR,
+                                (uint8_t *)&ucReadVal);
+    TEST_ASSERT_EQUAL(IOT_SDIO_SUCCESS, lRetVal);
+    TEST_ASSERT_EQUAL(eFunc, testIotSdio_FUNC_SEL_MASK & ucReadVal);
+
+    /* Io function enable */
+    xPerFuncEnable.eFunc = eSdioFunction1;
+    xPerFuncEnable.bEnable = true;
+    lRetVal = iot_sdio_ioctl(xSdioHandle,
+                             eSDIOEnableIo,
+                             (void *)&xPerFuncEnable);
+    TEST_ASSERT_EQUAL(IOT_SDIO_SUCCESS, lRetVal);
+
+    /* Verify io function enable */
+    lRetVal = iot_sdio_io_read_direct(xSdioHandle,
+                                eSdioFunction0,
+                                testIotSdio_IO_EN_ADDR,
+                                (uint8_t *)&ucReadVal);
+    TEST_ASSERT_EQUAL(IOT_SDIO_SUCCESS, lRetVal);
+    TEST_ASSERT_EQUAL(testIotSdio_IO1_EN_MASK, testIotSdio_IO1_EN_MASK & ucReadVal);
+
+    /* Io function disable */
+    xPerFuncEnable.eFunc = eSdioFunction1;
+    xPerFuncEnable.bEnable = false;
+    lRetVal = iot_sdio_ioctl(xSdioHandle,
+                             eSDIOEnableIo,
+                             (void *)&xPerFuncEnable);
+    TEST_ASSERT_EQUAL(IOT_SDIO_SUCCESS, lRetVal);
+
+    /* Verify io function disable */
+    lRetVal = iot_sdio_io_read_direct(xSdioHandle,
+                                eSdioFunction0,
+                                testIotSdio_IO_EN_ADDR,
+                                (uint8_t *)&ucReadVal);
+    TEST_ASSERT_EQUAL(IOT_SDIO_SUCCESS, lRetVal);
+    TEST_ASSERT_EQUAL(0, testIotSdio_IO1_EN_MASK & ucReadVal);
+
+
+    /* Set card inactive */
+    lRetVal = iot_sdio_ioctl(xSdioHandle,
+                             eSDIOSetCardInactive,
+                             NULL);
+    TEST_ASSERT_EQUAL(IOT_SDIO_SUCCESS, lRetVal);
+
+    if (btestIotSdioVerifyInactive) {
```

Comment:
q: Why is this guarded with a variable?

Reply:
Some card respond to the command with SUCCESS response but never go into inactive state. This variable allows config the DUT to go through this check or skip.